pkg/sentry/sentry.go

```diff
@@ -104,60 +108,96 @@ func (s *sentry) Start(ctx context.Context) error {
 		return fmt.Errorf("error creating security: %s", err)
 	}
 
-	val, err := s.validator(ctx)
+	vals, err := s.getValidators(ctx)
 	if err != nil {
 		return err
 	}
 
-	validatorErr := make(chan error, 1)
-	go func() {
-		log.Info("Starting validator")
-		validatorErr <- val.Start(ctx)
-	}()
-
-	log.Info("Validator created, starting sentry server")
-
-	providerErr := make(chan error, 1)
+	errCh := make(chan error, 2+len(vals))
+	for name, val := range vals {
+		go func(name config.ValidatorName, val validator.Validator) {
+			log.Infof("Starting validator %s", name)
+			errCh <- val.Start(ctx)
+		}(name, val)
+	}
 	go func() {
-		log.Info("starting security provider")
-		providerErr <- provider.Start(ctx)
+		log.Info("Starting security provider")
+		errCh <- provider.Start(ctx)
 	}()
 
 	sec, err := provider.Handler(ctx)
 	if err != nil {
-		return errors.Join(err, <-providerErr, <-validatorErr)
+		return err
 	}
 
-	log.Info("Security ready")
+	go func() {
+		log.Info("Starting Sentry server")
+		errCh <- server.Start(ctx, server.Options{
+			Port:             s.conf.Port,
+			Security:         sec,
+			Validators:       vals,
+			DefaultValidator: s.conf.DefaultValidator,
+			CA:               camngr,
+		})
+	}()
 
-	err = server.Start(ctx, server.Options{
-		Port:      s.conf.Port,
-		Security:  sec,
-		Validator: val,
-		CA:        camngr,
-	})
-	return errors.Join(err, <-providerErr, <-validatorErr)
+	// On the first error, cancel the context and return
+	return <-errCh
```

Comment:
We will only be returning the first error here, regardless if there were 2 or more.

Since we also need to propagate a closed context if _any_ routine returns error, we should use the `concurrency.RunnerManager` here.

Reply:
I'm not sure I understand what you mean with `concurrency.RunnerManager`. Can you explain please?

Reply:
Using the [`RunnerManager`](https://github.com/dapr/dapr/blob/master/pkg/concurrency/runner.go#L38) to manage both the `provider` and `server`, ensuring that both return and that both errors are caught.

Reply:
Done